File: PrimeNumbers.vb
Public Class PrimeNumbers

    Dim first As Integer
    Dim last As Integer
    Dim progress As Integer
    Dim list_of_primes As New List(Of Integer)(New Integer() {2, 3, 5, 7})
    Dim index As Integer
    Dim square_root As Integer
    Dim do_next_iteration As Boolean
    Dim result As New List(Of Integer)

    ' Simulate a constructor
    Public Sub _init_(alpha As Integer, omega As Integer)
        first = alpha
        last = omega
        ' STEP 1:
        progress = 9

        square_root = 0
    End Sub

    ' Returns a list Of the desired Set Of prime numbers
    Public Function getPrimes() As List(Of Integer)

        ' STEP 2:
        For progress = progress To last Step 2

            do_next_iteration = False ' a flag

            ' STEPS 3, 4:
            ' Check through already accumulated prime numbers
            ' to see if any Is a factor of "progress".
            square_root = Val(Math.Ceiling(Math.Sqrt(progress)))

            index = 0
            Do While list_of_primes.Item(index) <= square_root
                If progress Mod list_of_primes.Item(index) = 0 Then
                    do_next_iteration = True
                    Exit Do
                End If

                index += 1
            Loop

            If do_next_iteration = True Then
                Continue For
            End If

            ' if all checks are scaled,then "progress" Is our guy.
            list_of_primes.Add(progress)

        Next progress

        Return list_of_primes

    End Function

End Class

File: PrimeNumbers.py
from math import sqrt, ceil

# A class
class ListPrimes:

    # Constructor: takes two arguments  -- besides its name
    # alpha and omega values for the range
    def __init__(self, alpha, omega):
        self.start = alpha
        self.stop = omega + 1
        # STEP 1:
        self.progress = 9

        self.list_of_primes = [2, 3, 5, 7]
        self.square_root = 0


    # Returns an list of the desired set of prime numbers
    def getPrimes(self):

        # STEP 2:
        for self.progress in range(self.progress, self.stop, 2):

            self.do_next_iteration = False # a flag

            # STEPS 3, 4:
            # Check through already accumulated prime numbers
            # to see if any is a factor of "progress".
            self.square_root = int(ceil(sqrt(self.progress)))
            
            self.index = 0
            while self.list_of_primes[self.index] <= self.square_root:
                if self.progress % self.list_of_primes[self.index] == 0:
                    self.do_next_iteration = True
                    break

                self.index += 1

            if self.do_next_iteration == True:
                continue

            # if all checks are scaled,then "progress" is our guy.
            self.list_of_primes.append(self.progress)

        return self.list_of_primes
    
